kissnet.hpp

```diff
@@ -762,6 +783,31 @@ namespace kissnet
 
 				return static_cast<bool>(syscall_connect(sock, reinterpret_cast<SOCKADDR*>(&socket_output), sizeof(SOCKADDR)) != SOCKET_ERROR);
 			}
+#ifdef KISSNET_USE_OPENSSL
+			else if constexpr (sock_proto == protocol::tcp_ssl) //only TCP is a connected protocol
+			{
+				memcpy(&socket_output, getaddrinfo_results->ai_addr, sizeof(SOCKADDR));
+
+				if (!(static_cast<bool>(syscall_connect(sock, reinterpret_cast<SOCKADDR*>(&socket_output), sizeof(SOCKADDR)) != SOCKET_ERROR)))
+					return false;
+
+				auto *ssl_method = TLSv1_2_client_method();
+
+				pContext = SSL_CTX_new(ssl_method);
+				ssl = SSL_new(pContext);
+				if (!ssl)
+					return false;
+
+				if (!(static_cast<bool>(SSL_set_fd(ssl, sock))))
+					return false;
+
+				auto err = SSL_connect(ssl);
```

Comment:
If we don't do anything with the `err` value beside returning false, maybe just do 

```cpp
if(SSL_connect(ssl) <= 0)
    return false;
```

Reply:
Done!